iOS/TwitterImagePipeline/source/TwitterImagePipeline/ApiDefinition.cs

```diff
@@ -0,0 +1,2147 @@
+		[Field("TIPImagePipelineDidTearDownImagePipelineNotification", "__Internal")]
+		NSString TIPImagePipelineDidTearDownImagePipelineNotification { get; }
+	}
+
+	[Static]
+	partial interface TIPImagePipelineNotificationKeys
+	{
+		// extern NSString *const _Nonnull TIPImagePipelineImageIdentifierNotificationKey;
+		[Field("TIPImagePipelineImageIdentifierNotificationKey", "__Internal")]
+		NSString ImageIdentifier { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImageURLNotificationKey;
+		[Field("TIPImagePipelineImageURLNotificationKey", "__Internal")]
+		NSString ImageUrl { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImageDimensionsNotificationKey;
+		[Field("TIPImagePipelineImageDimensionsNotificationKey", "__Internal")]
+		NSString ImageDimensions { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImageContainerNotificationKey;
+		[Field("TIPImagePipelineImageContainerNotificationKey", "__Internal")]
+		NSString ImageContainer { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImageWasManuallyStoredNotificationKey;
+		[Field("TIPImagePipelineImageWasManuallyStoredNotificationKey", "__Internal")]
+		NSString ImageWasManuallyStored { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImagePipelineIdentifierNotificationKey;
+		[Field("TIPImagePipelineImagePipelineIdentifierNotificationKey", "__Internal")]
+		NSString ImagePipelineIdentifier { get; }
+
+		// extern NSString *const _Nonnull TIPImagePipelineImageTreatAsPlaceholderNofiticationKey;
+		[Field("TIPImagePipelineImageTreatAsPlaceholderNofiticationKey", "__Internal")]
+		NSString ImageTreatAsPlaceholder { get; }
+	}
+
+	// @interface TIPImagePipeline : NSObject
+	[BaseType(typeof(NSObject), Name = "TIPImagePipeline")]
+	[DisableDefaultCtor]
+	interface TIPImagePipeline : INativeObject
+	{
+		// @property (copy, atomic) NSArray<id<TIPImageAdditionalCache>> * _Nullable additionalCaches;
+		[NullAllowed, Export("additionalCaches", ArgumentSemantic.Copy)]
+		ITIPImageAdditionalCache[] AdditionalCaches { get; set; }
+
+		// @property (atomic) id<TIPImagePipelineObserver> _Nullable observer;
+		[NullAllowed, Export("observer", ArgumentSemantic.Assign)]
+		ITIPImagePipelineObserver Observer { get; set; }
+
+		// @property (readonly, copy, nonatomic) NSString * _Nonnull identifier;
+		[Export("identifier")]
+		string Identifier { get; }
+
+		// -(instancetype _Nullable)initWithIdentifier:(NSString * _Nonnull)identifier __attribute__((objc_designated_initializer));
+		[Export("initWithIdentifier:")]
+		[DesignatedInitializer]
+		IntPtr Constructor(string identifier);
+
+		// -(TIPImageFetchOperation * _Nonnull)operationWithRequest:(id<TIPImageFetchRequest> _Nonnull)request context:(id _Nullable)context delegate:(id<TIPImageFetchDelegate> _Nullable)delegate __attribute__((objc_requires_super));
+		[Export("operationWithRequest:context:delegate:")]
+		//[RequiresSuper] // TODO
```

Comment:
I am not 100% sure what to do here.  I am only getting one Google hit for this Attribute.  It looks like you should add an Advice attribute telling the user to call the base class?

https://bugzilla.xamarin.com/show_bug.cgi?id=58350

`[Advice ("You must call the base method when overriding.")]`